File: javascript/models/review.js
const Review = (function (){
  var id = 0
  return class Review{
    constructor(customer, restaurant){
      this.customer = customer
      customer.add_review(this)
      this.restaurant = restaurant
      this.id = ++id
      store.reviews.push(this)
    }
  }
}())

Review.prototype.all = function() {
  return store.reviews
};

File: object_relations/app/models/review.rb
class Review

  @@all = []

  def initialize(review, customer, restaurant)
    @review = review
    @restaurant = restaurant
    @customer = customer
    @customer.add_restaurant(@restaurant)
    @customer.add_review(self, @restaurant)
    @@all << self
  end

  def customer
    @customer
  end

  def restaurant
    @restaurant
  end

end
